proposal-EDGI_ML_project.md

```diff
@@ -0,0 +1,156 @@
+
+I plan to follow the following tentative timeline:
+
+- Community bonding period / Before May 30:
+  - Understand the differences in the data format of the different sources (PageFreezer, Versionista).
+  - Understand and contribute to the development of the ETL pipeline, as steps 0 and 1 are closely related to ETL pipeline design steps.
+- May 30 - June 12:
+  - Depending on the development status of the ETL pipeline, complete the creation of an initial training dataset in no more than two weeks.
+- June 13 - June 23:
+  - Work on the scripts of different Machine Learning models. Create a common script for loading data. Build the basic structure for different models.
+- June 24 - June 30:
+  - Summarize work, document it, add pull requests, and prepare for Phase 1 evaluations.
+- July 1 - July 14:
+  - Complete code for different models and train the models on the data.
+  - Based on the performance metrics, try to improve the performance and experiment with ensemble models.
+  - Start with the creation of visualizations and other reports based on the results of the trained models.
+- July 15 - July 22:
+  - Complete the reports and visualizations.
+  - Work on the serialization methods to store trained models for future use.
+- July 23 - July 28:
+  - Summarize work, document it, add pull requests, and prepare for Phase 2 evaluations.
+- July 29 - Aug 6:
+  - Retrieve the stored models using deserialization and test on new data.
+  - If required, simulate some test cases&#39; data and check the performance of models.
+- Aug 7 - Aug 10:
+  - Send the list of important changes to analysts for review.
+  - Create a mechanism to automatically add correctly prioritized changes to the training dataset once a minimum number of correct results is collected.
+- Aug 11 - Aug 21:
+  - My semester will start in the last week of July. I may have some time constraints in August. If the tasks planned for August take longer than expected, they will be completed in this period.
+  - Any issues raised on work completed before this will be tackled and if time permits, solved within this period.
+- Aug 22 - Aug 29:
+  - Summarize work, add pull requests for any remaining work, finish the documentation and get ready for final evaluations.
+
+I will like to continue working with EDGI after GSoC and be a part of the #DataRescue movement.
+
+**Deliverables** :
+
+EDGI will have a robust solution to the problem faced by analysts. The solution will reduce the effort it takes to find an important change (a needle) in a vast amount of changes (haystack).
+
+There will be 2 main deliverables:
+
+1. A solution to filter out the insignificant changes and avoid spending time on reviewing them.
+2. A base structure to get important changes in one place and use this for future tasks or projects.
```

Comment:
I have a simple separate table in mind. This can be added to the database. This will help us to quickly access reviewed changes and use them in the future.